src/dreyfus_index.erl

```diff
@@ -222,6 +243,35 @@ code_change(_OldVsn, State, _Extra) ->
 
 % private functions.
 
+get_index_with_sig(#doc{body={Fields}}=Doc, Sig) ->
+    RawIndexes = couch_util:get_value(<<"indexes">>, Fields, {[]}),
+    case RawIndexes of
+        {IndexList} when is_list(IndexList) ->
```

Comment:
this case statement has one clause, and crashes on anything else. If that's the intent, you don't need a case statement.